### ISSUE TITLE
NAs in assay matrix after mergeData

### ISSUE THREAD

**Issue by sdgamboa** [waldronlab/curatedMetagenomicData#234]
**Describe the bug**
NAs in the assay matrix after applying the `mergeData` function to a list of TreeSummarizedExperiments.

**To Reproduce**
``` r
suppressMessages(library(curatedMetagenomicData))
hmp_tse_list <- curatedMetagenomicData(".+HMP.+relative.abundance", dryrun = FALSE)
hmp_tse <- mergeData(hmp_tse_list)
sum(is.na(assay(hmp_tse)))
#> [1] 748112
```

<sup>Created on 2021-05-12 by the [reprex package](https://reprex.tidyverse.org) (v2.0.0)</sup>

**Expected behavior**
``` r
suppressMessages(library(curatedMetagenomicData))
hmp_tse_list <- curatedMetagenomicData(".+HMP.+relative.abundance", dryrun = FALSE)
#> snapshotDate(): 2021-05-05
hmp_tse <- mergeData(hmp_tse_list)
sum(is.na(assay(hmp_tse)))
#> [1] 0
```

**Additional context**
I'm not sure if this would be considered a bug or if it's important to distinguish between missing data and 0.

**Comment by lwaldron** [comment]
There should be no NA values in the assay data, just zeros and positive numeric values. Good catch @sdgamboa

**Comment by lwaldron** [comment]
As additional context, individual-sample profiles do not include any zero features, individual datasets merged from single samples have filled in zeros where a feature was missing from a single-sample profile (because it is zero, not unknown), and merged datasets should do the same (fill in zeros where a feature was missing).

**Comment by schifferl** [comment]
closed by 49a67ac
